src/celeritas/em/distribution/WentzelDistribution.hh

```diff
@@ -0,0 +1,387 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/em/distribution/WentzelDistribution.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Macros.hh"
+#include "corecel/Types.hh"
+#include "corecel/math/Algorithms.hh"
+#include "celeritas/em/data/WentzelData.hh"
+#include "celeritas/em/interactor/detail/PhysicsConstants.hh"
+#include "celeritas/em/xs/MottXsCalculator.hh"
+#include "celeritas/em/xs/WentzelXsCalculator.hh"
+#include "celeritas/mat/IsotopeView.hh"
+#include "celeritas/phys/ParticleTrackView.hh"
+#include "celeritas/random/distribution/BernoulliDistribution.hh"
+#include "celeritas/random/distribution/GenerateCanonical.hh"
+#include "celeritas/random/distribution/UniformRealDistribution.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Helper class for \c WentzelInteractor .
+ *
+ * Samples the scattering direction for the Wentzel model.
+ *
+ * References:
+ * [Fern] J.M. Fernandez-Varea, R. Mayol and F. Salvat. On the theory
+ *        and simulation of multiple elastic scattering of electrons. Nucl.
+ *        Instrum. and Method. in Phys. Research B, 73:447-473, Apr 1993.
+ *        doi:10.1016/0168-583X(93)95827-R
+ * [LR15] C. Leroy and P.G. Rancoita. Principles of Radiation Interaction in
+ *        Matter and Detection. World Scientific (Singapore), 4rd edition,
+ *        2015.
+ * [PRM]  Geant4 Physics Reference Manual (Release 11.1) sections 8.2 and 8.5
+ */
+class WentzelDistribution
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using MomentumSq = units::MevMomentumSq;
+    using Energy = units::MevEnergy;
+    using Mass = units::MevMass;
+    //!@}
+
+  public:
+    //! Construct with state and date from WentzelInteractor
+    inline CELER_FUNCTION
+    WentzelDistribution(ParticleTrackView const& particle,
+                        IsotopeView const& target,
+                        WentzelElementData const& element_data,
+                        real_type cutoff_energy,
+                        WentzelRef const& data);
+
+    //! Sample the scattering direction
+    template<class Engine>
+    inline CELER_FUNCTION Real3 operator()(Engine& rng) const;
+
+    //! Calculates the Moliere screening coefficient for the given target
+    inline CELER_FUNCTION real_type compute_screening_coefficient() const;
```

Comment:
Rename `compute_*` and `calculate_*` to `calc_` (or optionally drop the prefix if you wish for private functions) to be consistent with other parts of the code.